eth/precompiles/modexp.py

```diff
@@ -121,18 +126,19 @@ def _modexp(data: bytes) -> int:
     return result
 
 
-def modexp(computation: BaseComputation) -> BaseComputation:
+@curry
+def modexp(computation: BaseComputation,
+           gas_cost: Callable[..., int] = _compute_modexp_gas_fee_eip_198) -> BaseComputation:
```

Comment:
Maybe `gas_cost` -> `gas_calculator` or something to make it sound a little more method-y? Also, I think the type can be more tightly bound here.

```suggestion
           gas_calculator: Callable[[bytes], int] = _compute_modexp_gas_fee_eip_198) -> BaseComputation:
```